src/gc.c

```diff
@@ -3636,10 +3269,18 @@ void jl_init_thread_heap(jl_ptls_t ptls)
     gc_cache->perm_scanned_bytes = 0;
     gc_cache->scanned_bytes = 0;
     gc_cache->nbig_obj = 0;
-    size_t init_size = 1024;
-    gc_cache->pc_stack = (void**)malloc_s(init_size * sizeof(void*));
-    gc_cache->pc_stack_end = gc_cache->pc_stack + init_size;
-    gc_cache->data_stack = (jl_gc_mark_data_t *)malloc_s(init_size * sizeof(jl_gc_mark_data_t));
+
+    // Initialize GC mark-queue
+    size_t init_size = (1 << 18);
```

Comment:
2 MB seems rather large (though 8k does seem a little small)

Reply:
I think the chunk-queue shouldn't be large (the number of chunks grows with the depth of objarrays).

Reply:
This is over 2 MB per thread though, which seems large